src/rulesets/Base/arraymath.jl

```diff
@@ -94,6 +94,77 @@ function rrule(
     return A * B, times_pullback
 end
 
+#####
+##### fused 3-argument *
+#####
+
+using LinearAlgebra: mat_mat_scalar, mat_vec_scalar, rmul!, StridedMaybeAdjOrTransMat
+
+function rrule(
+    ::typeof(mat_mat_scalar),
+    A::StridedMaybeAdjOrTransMat{<:CommutativeMulNumber},
+    B::StridedMaybeAdjOrTransMat{<:CommutativeMulNumber},
+    γ::CommutativeMulNumber
+)
+    AB  = mat_mat_scalar(A, B, one(γ))
+    function mat_mat_scalar_back(Ȳ)
+        return (
+            NO_FIELDS,
+            InplaceableThunk(
+                @thunk(mat_mat_scalar(Ȳ, B', conj(γ))),
+                dA -> mul!(dA, Ȳ, B', conj(γ), true)
+            ),
+            InplaceableThunk(
+                @thunk(mat_mat_scalar(A', Ȳ, conj(γ))),
+                dB -> mul!(dB, A', Ȳ, conj(γ), true)
+            ),
+            @thunk if iszero(γ)
+                dot(AB, Ȳ)
+            else
+                dot(AB, Ȳ) / conj(γ)  # re-use mutated AB
```

Comment:
I don't get this comment, could you please elaborate?

Reply:
The gradient here should be `dot(A*B, dY)`, but at the time this gets run, it has done `rmul!(AB, γ)` and so the variable AB contains `A*B*γ`.

Reply:
Now simplified to just re-calculate `A*B` in the case when `γ==0`. That's a waste but surely that case is rare? And it's less confusing.